test/Sdk.Generator.Tests/FunctionMetadataProviderGeneratorTests.cs

```diff
@@ -0,0 +1,1066 @@
+            });
+            return builder;
+        }
+    }
+}
+".Replace("'", "\"");
+
+            await TestHelpers.RunTestAsync<FunctionMetadataProviderGenerator>(
+                referencedExtensionAssemblies,
+                inputCode,
+                expectedGeneratedFileName,
+                expectedOutput);
+        }
+
+        [Fact]
+        public async Task GenerateFunctionWhereOutputBindingIsInTheReturnTypeTest()
+        {
+            // test generating function metadata for a simple HttpTrigger
+            string inputCode = @"
+            using System.Net;
+            using Microsoft.Azure.Functions.Worker;
+            using Microsoft.Azure.Functions.Worker.Http;
+
+            namespace FunctionApp
+            {
+                public static class HttpTriggerWithMultipleOutputBindings
+                {
+                    [Function(nameof(HttpTriggerWithMultipleOutputBindings))]
+                    public static MyOutputType Run([HttpTrigger(AuthorizationLevel.Anonymous, ""get"", ""post"", Route = null)] HttpRequestData req,
+                        FunctionContext context)
+                    {
+                        var response = req.CreateResponse(HttpStatusCode.OK);
+                        response.WriteString(""Success!"");
+
+                        return new MyOutputType()
+                        {
+                            Name = ""some name"",
+                            HttpResponse = response
+                        };
+                    }
+                }
+
+                public class MyOutputType
+                {
+                    [QueueOutput(""functionstesting2"", Connection = ""AzureWebJobsStorage"")]
+                    public string Name { get; set; }
+
+                    public HttpResponseData HttpResponse { get; set; }
+                }
+            }";
+
+
+            string expectedGeneratedFileName = $"GeneratedFunctionMetadataProvider.g.cs";
+            string expectedOutput = @"// <auto-generated/>
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Core;
+using Microsoft.Azure.Functions.Worker.Core.FunctionMetadata;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+namespace Microsoft.Azure.Functions.Worker
+{
+    public class GeneratedFunctionMetadataProvider : IFunctionMetadataProvider
+    {
+        public Task<ImmutableArray<IFunctionMetadata>> GetFunctionMetadataAsync(string directory)
+        {
+            var metadataList = new List<IFunctionMetadata>();
+            var HttpTriggerWithMultipleOutputBindingsRawBindings = new List<string>();
+            var HttpTriggerWithMultipleOutputBindingsreqBinding = new {
+                name = 'req',
+                type = 'HttpTrigger',
+                direction = 'In',
+                authLevel = '((AuthorizationLevel)0).ToString();',
```

Comment:
authLevel value seems to be incorrect. It should not be string.  May be remove the quotes (in the generator) so the C# expression gets executed and produce the correct enum value.